### PR TITLE
[#IP-299] Add SubscriptionFeed Update on ServicePreference upsert

### PR DESCRIPTION
**Pull request by AleDore**
In order to complete the opt-in/opt-out service model implemetation, this PR adds subscription feed update while servicePreferences get upserted.
#### List of Changes

- `UpdateSubscriptionFeedActivity` refactor, in order to be called either from `UpsertedProfileOrchestrator` and `UpsertServicePreferences`
- Update `UpsertServicePreferences`'s handler
- `subscription_feed` utility methods moved apart from handlers functions

#### Motivation and Context

Every change made to servicePreference must be registered in subscriptionFeed in order to be consistent while sending notification to users.

#### How Has This Been Tested?

Unit test

#### Types of changes

- [ ] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: UpsertServicePreferences/subscription_feed.ts [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+          context,
+          input,
+          tableService,
+          subscriptionFeedTable
+        ),
+      toError
+    )
+  )
+    .chain(result => tryCatch(() => result, toError))
+    .foldTaskEither(
```
Comment (BurnedMarshal) [inline UpsertServicePreferences/subscription_feed.ts:76]: Here a more compact version
```suggestion
  tryCatch(
    () =>
      updateSubscriptionFeed(
        context,
        input,
        tableService,
        subscriptionFeedTable
      ),
    toError
  ).foldTaskEither(
```

File: UpsertServicePreferences/handler.ts [hunk 2]
```diff
@@ -202,18 +314,24 @@ export const GetUpsertServicePreferencesHandler = (
  * Wraps a UpsertServicePreferences handler inside an Express request handler.
  */
 export function UpsertServicePreferences(
+  telemetryClient: ReturnType<typeof initAppInsights>,
   profileModels: ProfileModel,
   serviceModels: ServiceModel,
-  servicePreferencesModel: ServicesPreferencesModel
+  servicePreferencesModel: ServicesPreferencesModel,
+  tableService: TableService,
+  subscriptionFeedTable: NonEmptyString
```
Comment (BurnedMarshal) [inline UpsertServicePreferences/handler.ts:322]:
```suggestion
  subscriptionFeedTableName: NonEmptyString
```

File: UpsertServicePreferences/handler.ts [hunk 3]
```diff
@@ -150,15 +173,43 @@ const upsertUserServicePreferences = (
...
  * Return a type safe GetServicePreferences handler.
  */
 export const GetUpsertServicePreferencesHandler = (
+  telemetryClient: ReturnType<typeof initAppInsights>,
   profileModels: ProfileModel,
   serviceModels: ServiceModel,
-  servicePreferencesModel: ServicesPreferencesModel
+  servicePreferencesModel: ServicesPreferencesModel,
+  tableService: TableService,
+  subscriptionFeedTable: NonEmptyString,
```
Comment (BurnedMarshal) [inline UpsertServicePreferences/handler.ts:209]:
```suggestion
  subscriptionFeedTableName: NonEmptyString,
```

File: package.json [hunk 4]
```diff
@@ -61,7 +61,7 @@
     "@pagopa/io-functions-commons": "^20.6.6",
     "@pagopa/ts-commons": "^9.6.0",
     "abort-controller": "^3.0.0",
-    "applicationinsights": "^1.7.4",
+    "applicationinsights": "^1.8.10",
```
Comment (BurnedMarshal) [inline package.json:64]: Why this change?

**Review by BurnedMarshal — CHANGES_REQUESTED** [review]

File: UpsertServicePreferences/subscription_feed.ts (hunk 1 above)
Comment (AleDore) [reply UpsertServicePreferences/subscription_feed.ts:76]: Since `updateSubscriptionFeed` can throw I've figured out that `tryCatch` handle Promise's resolve or reject but if the function throw it doesn't handle it :)

File: package.json (hunk 4 above)
Comment (AleDore) [reply package.json:64]: Because there are other dependencies that carry out `applicationinsights` ' `1.8.10` version, so I've updated it, since version `1.7.4` caused a mismatch of telemetry Client type. It was `NodeClient | DefaultClient` instead of simply `NodeClient`

**Review by BurnedMarshal — APPROVED** [review]
LGTM
